test/suites/rt-upgrade-chopsticks-orchestrator/test-upgrade-chain.ts

```diff
@@ -85,5 +96,77 @@ describeSuite({
                 expect(balanceBefore < balanceAfter).to.be.true;
             },
         });
+
+        it({
+            id: "T4",
+            timeout: 60000,
+            title: "Skip blocks until session change",
+            test: async () => {
+                const queuedValidators = (await api.query.session.queuedKeys())
+                    .map(([account, _]) => account.toHuman())
+                    .sort();
+
+                const { finalBlockNumber } = await skipToSessionChange();
+
+                const newValidators = (await api.query.session.validators()).map((v) => v.toHuman()).sort();
+                expect(
+                    queuedValidators,
+                    "Queued validators should match new validators after session change"
+                ).toMatchObject(newValidators);
+
+                const blockHash = await api.rpc.chain.getBlockHash(finalBlockNumber);
+                const block = await api.rpc.chain.getBlock(blockHash);
+                const signedExtrinsics = block.block.extrinsics.filter((x) => x.isSigned);
+                expect(signedExtrinsics.length, "SessionChange block should never have extrinsics inside").toBe(0);
+
+                const blockEvents = await api.query.system.events();
+                const sessionChangeEvent = blockEvents
+                    .filter(({ event }) => api.events.session.NewSession.is(event))
+                    .map(({ event }) => api.events.session.NewSession.is(event) && event.data)[0];
+                expect(sessionChangeEvent, "Session change event should be emitted").toBeTruthy();
+
+                const newSessionIndex = (await api.query.session.currentIndex()).toNumber();
+                expect(sessionChangeEvent.sessionIndex.toNumber()).toBe(newSessionIndex);
+
+                //  Dancelight does not have collatorAssignment events
```

Comment:
it does, it is named tanssiCollatorAssignment though

Reply:
what we usually do is retrieve the pallet name beforeAll based of it including "light" or "box"

Reply:
fixed